### DISCUSSION TITLE
Sunshine on Windows ARM

### DISCUSSION THREAD

**Discussion by ghost** [LizardByte/community#367] [Sunshine] [answered]
I posted this on Reddit a few days ago, and someone there advised me to post this problem here. So, here I am:

I'm having trouble running Sunshine on Windows 11 ARM, a Surface Pro X SQ1. Interestingly, the portable version works, but the installable one doesn't.

The error I encountered was:

> [2023-12-27 01:37:40.225973] [0x00003db0] [info] Failed to apply config: boost::filesystem::create_directories: Access denied [system:5]: "C:\Program Files\Sunshine\config" "C:\Program Files\Sunshine\config"
> [2023-12-27 01:37:40.245819] [0x00003db0] [info] To relaunch Sunshine successfully, use the shortcut in the Start Menu. Do not run Sunshine.exe manually.

Despite the second message, using the shortcut doesn't solve the problem.

Since the portable version works, the problem seems to be related to not having writing rights. I tried running it as an admin and got a different error:

> [2023-12-27 01:43:06.845300] [0x000040dc] [info] OpenService() failed: 1060
> Waiting for Web UI to be ready..................................................................................

After a minute or so, the Web UI is launched but doesn't show anything.

---
On a different topic:

The portable version works, but it has a noticeable delay. Reducing the resolution helps, but doesn't solve the problem entirely.
I use Sunshine/Moonlight with other devices in my home, and it works perfectly, so I don't think it's a network problem.

**Comment by ReenigneArcher** [comment]
I'm surprised either work, unless Windows is using some kind of built in emulation to run it.

**Reply by ghost** [reply]
Yes, Windows emulates x86 softwares on 10 and 11, and x64 ones on 11.

**Reply by ReenigneArcher** [reply, marked as answer]
> On a different topic:
> 
> The portable version works, but it has a noticeable delay. Reducing the resolution helps, but doesn't > solve the problem entirely.
> I use Sunshine/Moonlight with other devices in my home, and it works perfectly, so I don't think it's a network problem.

If it's emulating, it just going to be slow... nothing you can do about that unless you compile Sunshine yourself.

GitHub doesn't give us access to ARM build servers, and Sunshine is not properly setup for cross compilation as of right now.

**Comment by cgutman** [comment]
I actually messed around with this a while back. The major problem is the hardware encoding API support, which is probably why your performance is terrible. The only thing worse than software encoding is software encoding under emulation 😉 

I was playing around with adding Media Foundation encoding support in #681 which should work on ARM, but it never really panned out. Media Foundation in FFmpeg is buggy and lacks the zero-copy hwaccel encoding support that QuickSync, AMF, NVENC, etc have. In theory, Media Foundation can accept D3D11 textures, but it needs to be plumbed up in FFmpeg first.

**Comment by stefanw1337** [comment]
Similar issue:
`To relaunch Sunshine successfully, use the shortcut in the Start Menu. Do not run Sunshine.exe manually.`

But it never added a start-menu shortcut for me, adding one myself gives same error. So I can't run it. Using Windows 10 and nVidia 3080 Ti.
